packages/site/src/routes/[dictionaryId]/entry/_EntryDisplay.svelte

```diff
@@ -108,8 +110,12 @@
           display={$t(`entry.va`, {default: 'Variant'})}
           on:valueupdate />
     {/if}
+
+    <InstantSearch dictionaryId={$dictionary.id} let:search>
```

Comment:
This component needs to be as fast as possible when rendering views server-side as this an entry view is commonly shared link. Something large like InstantSearch that's just used in the Dialect selection modal should not be loaded here as it will slow down every load of this page, client and server side. Does Algolia have a HTTP API that we can hit from the DialectModal to see what dialects are used in a dictionary? If so, let's do that. If not, we can just init the search in the modal itself. But really, the larger question is, "Should we be standardizing the dialects that a dictionary is using as opposed to letting users just type in anything?" but we don't have time to focus on that task right now.

Reply:
I just read this, thanks for that. I'll check if we can use an HTTP API.

Reply:
@jacob-8 I reviewed how component composition works in svelte and I created a temporal solution for the first question I asked. Now InstantSearch only loads in the modal and starts when you type in. However I will search if I can find an API to make this even more efficient though.